Reworked PUAH

Xeinaemm opened the pull request:
### Changes:
- Removed corpse feature - too many bills depend on it, which always break something, including mods.
- Removed inventory threshold settings for hauling and updated translations
- TPS/FPS optimized. 12,3k things handled in 4:50 min with 10 pawns
- Changed merge priority to the highest. It fixes annoying problems with a constant lack of space, which is often not the case.
- Removed logs in the hot path. 
- Added compatibility with Allow Tools, by design, with any storage mod.
- Moved defs to the right folders.

pkp24 commented:
Sounds good, just made that pr setting change. 

Also for the not returning home until their inventory is full, I think it'd be better to do a search from that point and pickup what they can and then head back. Otherwise you may get some weird stuff going on. It's one of the changes I made in my local test branch.

I'll checkout what you've got and see if I can translate some of those changes into what you reworked.

pkp24 commented:
I added back the logging to a file and added the improved logging I was using in my testing a few days ago. I fixed a couple bugs I noticed as well on the branch I'm working on.

https://github.com/pkp24/PickupAndHaul/tree/reworked-puah

Xeinaemm committed:
Reworked PUAH

Xeinaemm commented:
Updated description. Almost done and ready to test. The only major issue is haul prioritization. I squashed the PR, so it's better to create a new one and cherry-pick your changes.

pkp24 commented:
The only solution I found to the default hauling was to completely take over hauling decisions. and sounds good, I'll check out what you've got

Xeinaemm commented:
Fixed all problems, including hauling. HasJobOnThing must always be **true**.

Xeinaemm commented:
Feel free to use this PR in PUAH. I will 'fork' myself into a performance mod where some of these changes will be included, but without annoying dependencies - it's peak what you can achieve in PUAH and maintain backward compatibility.

Xeinaemm committed:
Update pipeline
